HHS/Head-Start-TTADP
# TypeScript adoption

## Description of change

- entrypoint changed from `nodemon babel-node src/index.js` to `tsx watch src/index.ts`
- TS support for jest via babel with `@babel/preset-typescript`: https://jestjs.io/docs/getting-started#using-typescript
- all instances of `module.exports` changed to use esm `export` syntax
- `@typescript-eslint` added to eslint presets & plugins, so you should not get eslint warnings for unexpected tokens, while still getting warnings for the usual things:

<img width="931" alt="Screen Shot 2023-01-19 at 1 23 30 PM" src="https://user-images.githubusercontent.com/17074357/213552197-bac8aa7b-f72c-45ba-856d-fedc8eeb3071.png">

you should get helpful TS-related eslint recommendations:

<img width="1026" alt="Screen Shot 2023-01-20 at 10 52 30 AM" src="https://user-images.githubusercontent.com/17074357/213772115-74cd70b2-2dbb-4973-9a90-1df3d9f4a9c2.png">

because of the way `tsc` is creating the built files, the structure of the `build` folder has slightly changed, so this change was also necessary:

https://github.com/HHS/Head-Start-TTADP/blob/24f5464ded44fb77a8d0116a8fbf16244e5f1043/src/app.js#L25

## How to test

- `yarn install`. you'll probably need to restart vscode afterwards to make sure it picks up typescript and the new eslint parser.
- make sure you can use TS syntax without eslint errors.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
